### PR TITLE
Fix Angular datepicker to keep up with $digest cycle

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Fixes glitch documented in https://lab.civicrm.org/dev/core/-/issues/3763#note_79840

Before
----------------------------------------
Doesn't immediately update form state after changing input

After
----------------------------------------
Fixed

### CODE REVIEW

File: ang/crmUi.js [hunk 1]
```diff
@@ -82,14 +82,17 @@
...
+              // Because change gets triggered from the $render function we could be either inside or outside the $digest cycle
+              $timeout(function() {
+                var requiredLength = 19;
+                if (scope.crmUiDatepicker && scope.crmUiDatepicker.time === false) {
+                  requiredLength = 10;
+                }
+                if (scope.crmUiDatepicker && scope.crmUiDatepicker.date === false) {
+                  requiredLength = 8;
+                }
+                ngModel.$setValidity('incompleteDateTime', !(element.val().length && element.val().length !== requiredLength));
```
Comment (seamuslee001) [inline ang/crmUi.js:94]: @colemanw how is element defined is this just something funky in the jquery / angular stuff

Comment (seamuslee001) [reply ang/crmUi.js:94]: sorry i see it now no worries

**Comment by MegaphoneJon** [comment]
It seems like there are 2 bugs causing the same problem here, and this fixes one.  I would still consider the requirement to fill in the time to be a bug, but without this patch, entering the date and time still doesn't enable **Save**.  So this should be MOP in my opinion.

**Comment by colemanw** [comment]
retest this please

**Pull request merged**